otter/log/cloudfeeds.py

```diff
@@ -0,0 +1,177 @@
+        "title": "autoscale",
+        "content": {
+            "event": {
+                "@type": "http://docs.rackspace.com/core/event",
+                "id": "",
+                "version": "2",
+                "eventTime": "",
+                "type": "INFO",
+                "region": "",
+                "product": {
+                    "@type": "http://docs.rackspace.com/event/autoscale",
+                    "serviceCode": "Autoscale",
+                    "version": "1",
+                    "message": ""
+                }
+            }
+        }
+    }
+}
+
+
+def prepare_request(req_fmt, event, error, timestamp, region):
+    """
+    Prepare request based on request format
+
+    :returns: Effect of `str` containing request
+    """
+    request = deepcopy(req_fmt)
+    if error:
+        request['entry']['content']['event']['type'] = 'ERROR'
+    request['entry']['content']['event']['region'] = region
+    request['entry']['content']['event']['eventTime'] = timestamp
+    request['entry']['content']['event']['product'].update(event)
+
+    def set_id(_id):
+        request['entry']['content']['event']['id'] = _id
+        return request
+
+    return Effect(Func(uuid.uuid4)).on(set_id)
+
+
+def add_event(event, tenant_id, region, log):
+    """
+    Add event to cloud feeds
+    """
+    event, error, timestamp = sanitize_event(event)
+    req_eff = prepare_request(request_format, event, error, timestamp, region)
+
+    def _send_event(req):
+        eff = retry_effect(
+            service_request(
+                ServiceType.CLOUD_FEEDS, 'POST',
+                append_segments('autoscale', 'events'),
+                data=req, log=log, success_pred=has_code(201)),
+            retry_times(5), exponential_backoff_interval(2))
+        return Effect(TenantScope(tenant_id=tenant_id, effect=eff))
+
+    return req_eff.on(_send_event)
+
+
+@attributes(['reactor', 'authenticator', 'tenant_id', 'region',
+             'service_configs', 'log', 'get_disp', 'add_event'],
+            defaults={'log': otter_log, 'get_disp': get_full_dispatcher,
+                      'add_event': add_event})
+class CloudFeedsObserver(object):
+    """
+    Log observer that pushes events to cloud feeds
+    """
+
+    def __call__(self, event_dict):
+        """
+        Process event and push it to Cloud feeds
+        """
+        if not event_dict.get('cloud_feed', False):
+            return
+        # Do further logging without cloud_feed to avoid coming back here
+        # in infinite recursion
+        log_keys = keyfilter(
+            lambda k: k not in ('message', 'cloud_feed'), event_dict)
+        log = self.log.bind(
+            system='otter.cloud_feed', cf_msg=event_dict['message'][0],
+            **log_keys)
+        try:
+            eff = self.add_event(event_dict, self.tenant_id, self.region, log)
+        except UnsuitableMessage as me:
+            log.err(None, ('Tried to add unsuitable message in cloud feeds: '
```

Comment:
You wanna add an `otter_event_type` key here? I suggest something like "unsuitable-cloud-feeds-message".